drivers/storage/portworx/component/windows.go

```diff
@@ -0,0 +1,263 @@
+		errList = append(errList, err.Error())
+	}
+
+	if err := w.createStorageClass(); err != nil {
+		logrus.Errorf("error during creating %s storageclass %s ", WindowsStorageClass, err)
+		errList = append(errList, err.Error())
+	}
+
+	if len(errList) > 0 {
+		return commonerrors.New(strings.Join(errList, " , "))
+	}
+	return nil
+}
+
+func (w *windows) Delete(cluster *corev1.StorageCluster) error {
+	var errList []string
+
+	if err := k8s.DeleteDaemonSet(w.client, WindowsCsiDaemonsetName, core.NamespaceSystem); err != nil {
+		errList = append(errList, err.Error())
+	}
+
+	if err := k8s.DeleteDaemonSet(w.client, WindowsInstallerDaemonsetName, cluster.Namespace); err != nil {
+		errList = append(errList, err.Error())
+	}
+
+	if cluster.DeletionTimestamp != nil &&
+		cluster.Spec.DeleteStrategy != nil && cluster.Spec.DeleteStrategy.Type != "" {
+		if err := k8s.DeleteStorageClass(w.client, WindowsStorageClass); err != nil {
+			errList = append(errList, err.Error())
+		}
+	}
+	if len(errList) > 0 {
+		return commonerrors.New(strings.Join(errList, " , "))
+	}
+
+	w.MarkDeleted()
+	return nil
+}
+
+func (w *windows) MarkDeleted() {
+	w.isCsiCreated = false
+	w.isInstallerCreated = false
+}
+
+// RegisterPortworxPluginComponent registers the PortworxPlugin component
+func RegisterWindowsComponent() {
+	Register(WindowsComponentName, &windows{})
+}
+
+func init() {
+	RegisterWindowsComponent()
+}
+
+func (w *windows) createDaemonSet(filename, daemonsetName, nameSpace string, cluster *corev1.StorageCluster) error {
+	daemonSet, err := k8s.GetDaemonSetFromFile(filename, pxutil.SpecsBaseDir())
+	if err != nil {
+		return err
+	}
+
+	daemonSet.Name = daemonsetName
+	daemonSet.Namespace = nameSpace
+	for i, container := range daemonSet.Spec.Template.Spec.Containers {
+		var image string
+		if container.Name == "node-driver-registrar" {
+			image = w.getDesiredNodeRegistrarImage(cluster)
+		}
+
+		if container.Name == "liveness-probe" {
+			image = w.getDesiredLivenessImage(cluster)
+		}
+
+		if container.Name == "windowsinstaller" {
+			image = w.getDesiredInstallerImage(cluster)
+		}
+
+		daemonSet.Spec.Template.Spec.Containers[i].Image = image
+	}
+
+	existingDaemonSet := &appsv1.DaemonSet{}
+	getErr := w.client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      daemonsetName,
+			Namespace: nameSpace,
+		},
+		existingDaemonSet,
+	)
+	if getErr != nil && !errors.IsNotFound(getErr) {
+		return getErr
+	}
+
+	pxutil.ApplyStorageClusterSettingsToPodSpec(cluster, &daemonSet.Spec.Template.Spec)
+	pxutil.ApplyWindowsSettingsToPodSpec(&daemonSet.Spec.Template.Spec)
+
+	equal, _ := util.DeepEqualPodTemplate(&daemonSet.Spec.Template, &existingDaemonSet.Spec.Template)
+	if (!w.isCsiCreated && daemonSet.Name == WindowsCsiDaemonsetName) ||
+		(!w.isInstallerCreated && daemonSet.Name == WindowsInstallerDaemonsetName) ||
+		!equal {
+		if err := k8s.CreateOrUpdateDaemonSet(w.client, daemonSet, nil); err != nil {
```

Comment:
Let's add the owner reference for the installer DS as it would be running in the cluster's namespace.

Reply:
added